### DISCUSSION TITLE
"Connection refused" while pinging DB during setup

### DISCUSSION THREAD

**Discussion by rtuite** [semaphoreui/semaphore#1700] [Q&A] [answered]
I am setting up a new Ansible instance and would like to use Semaphore to manage it. I've only used Ansible in a test environment to learn the basics, and I had this same issue with Semaphore there. 

I installed the package and ran `semaphore setup` to configure it. I accept all defaults (except the DB password obv) and when it gets to the point where it writes the configuration to `config.json` and pings the DB, it fails with the following error:

```
Configuration written to /home/<user redacted>/config.json..
 Pinging db..
WARN[0063] dial tcp 127.0.0.1:3306: connect: connection refused 
panic: dial tcp 127.0.0.1:3306: connect: connection refused
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8c8f37]

goroutine 1 [running]:
github.com/ansible-semaphore/semaphore/db/sql.(*SqlDb).Close(0x1fc1820?, {0xc0001dee38?, 0x4ae73b?})
        /home/runner/work/semaphore/semaphore/db/sql/SqlDb.go:251 +0x17
panic({0x96aec0, 0xc0004ce0f0})
        /opt/hostedtoolcache/go/1.18.7/x64/src/runtime/panic.go:838 +0x207
github.com/ansible-semaphore/semaphore/db/sql.(*SqlDb).Connect(0xc0000a8628, {0xc0000a8008?, 0xc0001df988?})
        /home/runner/work/semaphore/semaphore/db/sql/SqlDb.go:278 +0xb05
github.com/ansible-semaphore/semaphore/cli/cmd.doSetup()
        /home/runner/work/semaphore/semaphore/cli/cmd/setup.go:41 +0x17f
github.com/ansible-semaphore/semaphore/cli/cmd.glob..func4(0x22e7600?, {0x9d3fdc?, 0x0?, 0x0?})
        /home/runner/work/semaphore/semaphore/cli/cmd/setup.go:23 +0x17
github.com/spf13/cobra.(*Command).execute(0x22e7600, {0x2322918, 0x0, 0x0})
        /home/runner/work/semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x22e6700)
        /home/runner/work/semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/work/semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:902
github.com/ansible-semaphore/semaphore/cli/cmd.Execute()
        /home/runner/work/semaphore/semaphore/cli/cmd/root.go:36 +0x5b
main.main()
        /home/runner/work/semaphore/semaphore/cli/main.go:8 +0x17
```

I'm following the instructions for installing from a package manager located at https://docs.semui.co/administration-guide/installation#package-manager. I also verified that I have Python 3.10.12 and Git 2.34.1 installed. I found a few references to similar issues, but the most recent post was from 2018 and did not seem applicable to my scenario as the issue was that they were passing an invalid configuration after a validation was added to reading the IP address:port in the `config.json` file. This scenario had a previously functioning platform that broke with an update. I'm working with a clean install of Ubuntu server with only Ansible and Semaphore installed.

**Comment by rtuite** [comment, marked as answer]
I found the problem and it wasn't networking. The setup guide does not specify that you have to manually install your database platform and create the database user... I manually installed MySQL and set up the DB user and now it is working. This may be common knowledge to people experienced with these platforms, but I am not so it was not obvious for me.

**Reply by fiftin** [reply]
Hi @rtuite 
Database will be created automatically if db user has permissions for this.

**Reply by rtuite** [reply]
That is what I meant, I've corrected this.
